test_contextual_lenses/contextual_lenses.py

```diff
@@ -0,0 +1,128 @@
+"""Contextual lenses."""
+"""Creates sequence length independent representation of embedded sequences."""
+"""Original paper: https://arxiv.org/pdf/2002.08866.pdf."""
+
+
+import flax
+from flax import nn
+
+import jax
+import jax.nn
+import jax.numpy as jnp
+
+import numpy as np
+
+
+def max_pool(x, padding_mask=None, **reduce_fn_kwargs):
+  """Apply padding, take maximum over sequence length axis."""
+
+  if padding_mask is not None:
+    x = x * padding_mask
+    neg_mask = - 999999. * (1 - padding_mask)
+    x = x + neg_mask
+   
+  rep = jnp.max(x, axis=-2)
+   
+  return rep
+
+
+def mean_pool(x, padding_mask=None, **reduce_fn_kwargs):
+  """Apply padding, take mean over sequence length axis."""
+
+  if padding_mask is not None:
+    x = x * padding_mask
+    rep = jnp.sum(x, axis=-2) / jnp.sum(padding_mask, axis=-2)
+  else:
+    rep = jnp.mean(x, axis=-2)
+
+  return rep
+
+
+def linear_max_pool(x, padding_mask=None, **reduce_fn_kwargs):
```

Comment:
Here and below you should have a named argument rep_size. There's no reason to do the unpacking of generic kwargs on line 45.